pkg/iac/rego/load.go

```diff
@@ -325,8 +363,14 @@ func (s *Scanner) filterModules() error {
 			return fmt.Errorf("retrieve metadata for module %s: %w", name, err)
 		}
 
-		if s.isModuleApplicable(module, metadata, name) {
-			filtered[name] = module
+		if metadata != nil {
+			if !s.IsMinimumVersionSupported(metadata, module) {
+				continue
+			}
+
+			if s.isModuleApplicable(module, metadata, name) {
+				filtered[name] = module
+			}
```

Comment:
note: we should first check if metadata is not nil prior to checking for `isModuleApplicable` as fields of the metadata structure are used within this func which can lead to a panic if there's no metadata initialized. 

It also prevents a check to run without any metadata present. If such a check is loaded, we are not able to display the results for it as essential things such as IDs, Severity are missing if no metadata is present.